ReactWindows/ReactNative.Shared/Common/AccessibilityHelper.cs

```diff
@@ -0,0 +1,229 @@
+
+        /// <summary>
+        /// Sets the ImportantForAccessibility property for uiElement.
+        /// It uses AutomationProperties.AccessibilityView to implement the element and its children
+        /// visibility to narrator. ImportantForAccessibility value is stored as an attached property.
+        /// </summary>
+        /// <param name="uiElement"></param>
+        /// <param name="importantForAccessibilityNew"></param>
+        public static void SetImportantForAccessibility(UIElement uiElement, ImportantForAccessibility importantForAccessibilityNew)
+        {
+            var importantForAccessibilityOld = GetImportantForAccessibilityAttached(uiElement);
+            if (importantForAccessibilityOld == importantForAccessibilityNew)
+            {
+                return;
+            }
+
+            SetImportantForAccessibilityAttached(uiElement, importantForAccessibilityNew);
+
+            var uiElementAutomationPeer = FrameworkElementAutomationPeer.FromElement(uiElement);
+            // If this element is 'hidden' by a parrent, do not update AccessibilityView property.
+            if (IsHiddenElement(uiElementAutomationPeer))
+            {
+                return;
+            }
+
+            // Update AccessibilityView property accordingly.
+            switch (importantForAccessibilityNew)
+            {
+                case ImportantForAccessibility.Auto:
+                    uiElement.ClearValue(AutomationProperties.AccessibilityViewProperty);
+                    SetChildrenAccessibilityViewFromImportantForAccessibility(uiElementAutomationPeer);
+                    break;
+                case ImportantForAccessibility.No:
+                    AutomationProperties.SetAccessibilityView(uiElement, AccessibilityView.Raw);
+                    SetChildrenAccessibilityViewFromImportantForAccessibility(uiElementAutomationPeer);
+                    break;
+                case ImportantForAccessibility.NoHideDescendants:
+                    AutomationProperties.SetAccessibilityView(uiElement, AccessibilityView.Raw);
+                    SetChildrenAccessibilityView(uiElementAutomationPeer, AccessibilityView.Raw);
+                    break;
+                case ImportantForAccessibility.Yes:
+                    AutomationProperties.SetAccessibilityView(uiElement, AccessibilityView.Content);
+                    SetChildrenAccessibilityView(uiElementAutomationPeer, AccessibilityView.Raw);
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Attached property used to store ImportantForAccessibility value in native controls.
+        /// </summary>
+        public static readonly DependencyProperty ImportantForAccessibilityAttachedProperty = DependencyProperty.RegisterAttached(
+                                                                                                "ImportantForAccessibilityAttached",
```

Comment:
>                                                                                         [](start = 8, length = 88)

maybe use 4 space indent instead of this large indent?

Reply:
Fixed.

---
In reply to: [163091191](https://github.com/Microsoft/react-native-windows/pull/1595#discussion_r163091191) [](ancestors = 163091191)